homeassistant/components/weatherflow_cloud/entity.py

```diff
@@ -0,0 +1,43 @@
+"""Base entity class for WeatherFlow Cloud integration."""
+
+from homeassistant.helpers.device_registry import DeviceEntryType, DeviceInfo
+from homeassistant.helpers.entity import EntityDescription
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import ATTR_ATTRIBUTION, DOMAIN, MANUFACTURER
+from .coordinator import WeatherFlowCloudDataUpdateCoordinator
+
+
+def get_station_device_info(station_name: str, station_id: int) -> DeviceInfo:
+    """Generate attr_device_info from station name/id.."""
+    return DeviceInfo(
+        name=station_name,
+        entry_type=DeviceEntryType.SERVICE,
+        identifiers={(DOMAIN, f"{station_id}")},
+        manufacturer=MANUFACTURER,
+        configuration_url=f"https://tempestwx.com/station/{station_id}/grid",
+    )
+
+
+class WeatherFlowCloudEntity(CoordinatorEntity[WeatherFlowCloudDataUpdateCoordinator]):
+    """Base entity class to use for sensors."""
```

Comment:
Not only sensors right

Reply:
Its only for sensors right now ... because weather doesn't inherit from it ... which I should probably change